app/src/main/java/com/example/coronadiagnosticapp/data/db/entity/question/Question.kt

```diff
@@ -0,0 +1,53 @@
+package com.example.coronadiagnosticapp.data.db.entity.question
+
+import androidx.room.PrimaryKey
+import com.google.gson.annotations.SerializedName
+import java.util.*
+
+abstract class Question(
+    @PrimaryKey
+    val id: Long,
+    var name: String,
+    var displayName: String,
+    @SerializedName("qtype")
+    var type: QuestionType,
+    val addedOn: Date,
+    @SerializedName("required")
+    var isRequired: Boolean = true
+) {
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
+
+        other as Question
+
+        if (id != other.id) return false
+        if (name != other.name) return false
+        if (displayName != other.displayName) return false
+        if (type != other.type) return false
+        if (addedOn != other.addedOn) return false
+        if (isRequired != other.isRequired) return false
+
+        return true
+    }
+
+    override fun hashCode(): Int {
```

Comment:
This is something for Caching I guess?

Reply:
it's basically for maps , could be used for ordering and quick extraction
the hash is a unique id of the object